kernel/init.c

```diff
@@ -191,6 +182,15 @@ void z_data_copy(void)
 
 bool z_sys_post_kernel;
 
+void immediate_printk(const char *fmt, ...)
+{
+	va_list ap;
+
+	va_start(ap, fmt);
+	vprintk(fmt, ap);
+	va_end(ap);
+}
```

Comment:
I don't get it.  What is the point of this?

Reply:
printk() is calling log_printk() when CONFIG_LOG_PRINTK is on. Then the output would be happening in the logging thread which is halted while the busy-loop is running. The output of the delay would only happen after the delay and thus leaving the user wondering what is happening.

This way the message is outputted before the busy-loop executes and the user knows that boot is delayed.